src/System.Collections/src/System/Collections/Generic/HashSetEqualityComparer.cs

```diff
@@ -50,13 +87,10 @@ public override bool Equals(object obj)
             {
                 return false;
             }
-            return (_comparer == comparer._comparer);
-        }
 
-        public override int GetHashCode()
-        {
-            return _comparer.GetHashCode();
+            return _comparer?.Equals(comparer._comparer) ?? comparer._comparer == null;
```

Comment:
I probably also shouldn't depend on `_comparer` not considering itself equal to null, as strange as that should be.